### PR TITLE
Add test for on-static fileserver

### PR DESCRIPTION
**Pull request by nortonluo**
@cgx027 @yyscamper @anhou @chenge3 @sunnyqianzhang

### CODE REVIEW

**Comment by nortonluo** [comment]
WIP

**Comment by anhou** [comment]
@nortonluo maybe RackHD/RackHD repo is not the proper place to put these test codes.

**Comment by nortonluo** [comment]
@anhou The test script is a part of FIT test. FIT test is now merged to RackHD repo

File: test/fit_tests/tests/fileserver/test_microkernel_static_file_server.py [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+        api_data = fit_common.rackhdapi("/api/2.0/obms", action='put', payload=payload)
+        if api_data['status']== 201:
+            return True
+    return False
+
+
+# Select test group here using @attr
+from nose.plugins.attrib import attr
+@attr(all=True, regression=True, smoke=True)
+class static_microkernel_file_server(fit_common.unittest.TestCase):
```
Comment (hohene) [inline test/fit_tests/tests/fileserver/test_microkernel_static_file_server.py]: How long does this test take to run, does it meet smoke criteria?
And can it be run against virtual/vargrant environment?
if not, change smoke=False.
thanks!

Comment (nortonluo) [inline test/fit_tests/tests/fileserver/test_microkernel_static_file_server.py]: Good catch!  I will change that!

File: test/fit_tests/tests/fileserver/test_microkernel_static_file_server.py [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+
+# Local methods
+NODECATALOG = fit_common.node_select()
+
+
+
+def filerestful(url_command, rest_action='get', rest_payload=[], rest_timeout=None, sslverify=False, rest_headers={}):
+    '''
+    This function is a copy of the restful function used in fit_common. Due to the external file server used different authtication.
+    This routine executes a rest API call to the host.
```
Comment (nortonluo) [inline test/fit_tests/tests/fileserver/test_microkernel_static_file_server.py]: Initially I used fit_common to do this job but I then found there is two trouble:1. fit_common use ora IP to execute while the file server. And I need to access both servers in the bootstrapping test. 2. Static server do not have authentication but RackHD ora has authentication. I get trouble when call them together

File: test/fit_tests/tests/fileserver/README.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
+# External File Server(On-static) Test:
+
+This test validate the function of RackHD On-static file server. You need put an config file in the /config directory containing the test file url. The test date should write in jon format and save as fileserver_config.jon. The following is an example.
```
Comment (nortonluo) [inline test/fit_tests/tests/fileserver/README.md]: Good catch, Json, forgive my typo :)
Do you think should I upload the fileserver_config.json in to github? I worried there are some internal links/server in this file

**Comment by johren** [comment]
Checking in on older PRs.  It looks like the last activity was on 12/14.  Please update with further comment if you intend to move forward with this PR.  Otherwise, I will close on 2/18.

**Comment by nortonluo** [comment]
@anhou @yyscamper  @leoyjchang  As on-static is not mereged into RackHD repo. How do you think about deal with this PR? How about move it out to other repo like on-tools or so?

**Comment by nortonluo** [comment]
@johren pre discussed with the team, we thought this PR has dependence with on-static module which is still in private repo. Where the on-static will goes to is still TBD. I agree to close this PR first. And we can open another PR once the on-static is merged to public repo.

**Comment by johren** [comment]
Okay, thanks @nortonluo .  I will close.

**Pull request closed without merging**